compiler/docgen.nim

```diff
@@ -659,6 +659,7 @@ proc genOutFile(d: PDoc): Rope =
 proc generateIndex*(d: PDoc) =
   if optGenIndex in gGlobalOptions:
     writeIndexFile(d[], splitFile(options.outFile).dir /
+                        splitFile(d.filename).dir /
```

Comment:
I'm assuming that ``d.filename`` isn't an absolute path here? It may be worth adding an assert for this.